Singular/LIB/arc.lib

```diff
@@ -0,0 +1,571 @@
+        currentPoly = reducedBigIdeal[bigIdealIndex];
+        
+        for (basisIdx = 1; basisIdx < size(basis_JF_T); basisIdx++)
+        {
+            basisElement = basis_JF_T[basisIdx];
+            
+            coefMatrix = coef(currentPoly, basisElement);
+            
+            for (matrixColIndex = 1; matrixColIndex <= ncols(coefMatrix); matrixColIndex++)
+            {
+                if (reduce(coefMatrix[1,matrixColIndex], basisElement) == 0)
+                {
+                    if (nrows(coefMatrix) >= 2) {
+                        term = coefMatrix[2,matrixColIndex];
+                    } else {
+                        term = coefMatrix[1,matrixColIndex];
+                    }
+                    
+                    
+                    for (originalVarScanIndex = 1; originalVarScanIndex <= numbVars; originalVarScanIndex++) {
+                        if (deg(term, originalVarScanIndex) > 0) {
+                            // Extract all coefficients including constants
+                            coefMatrix = coeffs(term, var(originalVarScanIndex));
+                            poly result = 0;
+                            // Process all columns in the matrix to capture every coefficient
+                            for (int i = 1; i <= ncols(coefMatrix); i++) {
+                                result = result + coefMatrix[2,i];
+                            }
+                            term = result;
+                        }
+                    }
+                    
+                    shouldKeep = 1;
+                    if (useOptList == 1)
+                    {
+                        for (optIdx=1; optIdx<=size(optList); optIdx++)
+                        {
+                            varIdx = optList[optIdx];
+                            varToFind = "a"+string(varIdx)+"*";
+                            termStr   = string(term)+"*";
+                            if (find(termStr,varToFind)!=0)
+                            {
+                                shouldKeep = 0;
+                                break;
+                            }
+                        }
+                    }
+                    if (shouldKeep == 1)
+                    {
+                        arcideal = arcideal + term;
+                    }
+                }
+            }
+            
+            for (matrixColIndex=1; matrixColIndex<=ncols(coefMatrix); matrixColIndex++)
+            {
+                if (reduce(coefMatrix[1,matrixColIndex], basis_JF_T[basisIdx])==0 && nrows(coefMatrix)>=2)
+                {
+                    currentPoly = currentPoly - coefMatrix[2,matrixColIndex]*coefMatrix[1,matrixColIndex];
+                }
+            }
+        }
+        
+        if (currentPoly!=0 && currentPoly!=1)
+        {
+            
+            for (originalVarScanIndex = 1; originalVarScanIndex <= numbVars; originalVarScanIndex++) {
+                if (deg(currentPoly, originalVarScanIndex) > 0) {
```

Comment:
deg with 2 arguments expects deg(<poly>,<intvec>).
Here originalVarScanIndex (int) will be converted to intvec(originalVarScanIndex).
Maybe it should be
```
intvec ww;
for (...)
{
  ww=0:vars(basering);
  ww[originalVarScanIndex]=1;
  if(deg(currentPoly,ww)>0) ....
```